internal/drain_runner/runner.go

```
@@ -249,6 +251,19 @@ func (runner *drainRunner) checkPreprocessors(ctx context.Context, candidate *co
 	return
 }
 
+// resetPreProcessors will iterate over all pre processors and call the reset function.
+func (runner *drainRunner) resetPreProcessors(ctx context.Context, candidate *corev1.Node) {
+	span, ctx := tracer.StartSpanFromContext(ctx, "ResetPreProcessors")
+	defer span.Finish()
+
+	for _, pre := range runner.preprocessors {
+		err := pre.Reset(ctx, candidate)
+		if err != nil {
```

I was thinking about reporting that error up in the stack be realise that would be hard to manage because may have been reset but not others.

How should we handle these errors? nobody is going to cleanup the annotation, and we only have this log.
Should we report a metric and alert?

I am afraid that this will silently fail (probably not often) but it would be hard to back track.